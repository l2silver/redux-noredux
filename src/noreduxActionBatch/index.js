// @flow
import noreduxAction from '../noreduxAction'

export default function noreduxActionBatch(reducers: Object) {
  return Object.keys(reducers).reduce((finalResult, key) => {
    finalResult[key] = (...args) => noreduxAction(reducers[key](...args))
    return finalResult
  }, {})
}
