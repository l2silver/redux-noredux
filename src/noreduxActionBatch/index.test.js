// @flow
import scopeReducers from 'noredux/lib/scopeReducers'
import noreduxActionBatch from './'


describe('noreduxActionBatch', ()=>{
  it('returns actions', ()=>{
    const reducers = {
      create: (payload)=>(state)=>(payload),
    };
    const noreduxReducers = noreduxActionBatch(reducers)
    expect(noreduxReducers.create(1)).toMatchSnapshot()
  })
  it('returns actions with noredux wrappings', ()=>{
    const reducers = {
      create: (payload)=>(state)=>(payload),
    };
    const noreduxReducers = noreduxActionBatch(scopeReducers('users', reducers))
    expect(noreduxReducers.create(1)).toMatchSnapshot()
  })
})