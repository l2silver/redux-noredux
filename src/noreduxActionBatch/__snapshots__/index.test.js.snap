// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`noreduxActionBatch returns actions 1`] = `
Object {
  "args": undefined,
  "noredux": true,
  "reducer": [Function],
  "type": "@@redux-noredux/",
}
`;

exports[`noreduxActionBatch returns actions with noredux wrappings 1`] = `
Object {
  "args": Array [
    1,
  ],
  "noredux": true,
  "reducer": [Function],
  "type": "@@redux-noredux/users:create",
}
`;
